pkg/controller/jobs/trainjob/trainjob_controller.go

```diff
@@ -162,6 +165,9 @@ func (t *TrainJob) PodLabelSelector() string {
 }
 
 func getChildJobSet(t *TrainJob) (*jobsetapi.JobSet, error) {
+	getChildJobSetMutex.Lock()
+	defer getChildJobSetMutex.Unlock()
```

Comment:
Can you instead wrap just calling `runtime.RuntimeInfo`? For simplity you can move that invocation to another helper function

```golang
func runtimeInfo(t *TrainJob) (*info, error) {
   runtimeInfoMutex.Lock()
	defer runtimeInfoMutex.Unlock()
   return runtime.RuntimeInfo(trainJob, trSpec.Template, trSpec.MLPolicy, trSpec.PodGroupPolicy)
}
```